### PR TITLE
Feature 934/implementing cypress in the development environment

### PR DESCRIPTION
**Pull request by pieperm**
In this PR, Cypress was implemented and configured in the Check-ins application. It now runs as part of the Gradle build. 

If you'd like to see the Cypress tests run: use the command `npx Cypress run`. (Note: The test that runs is a dummy test - only to check the functionality of Cypress in the app and the Github workflow.)

### CODE REVIEW

File: .github/workflows/gradle-build-feature.yml [hunk 1]
```diff
@@ -27,6 +41,23 @@ jobs:
...
+        run: ./gradlew dockerBuild
+      - name: Run docker container
+        run: ./gradlew dockerRun
+      - name: Cypress runs tests
+        uses: cypress-io/github-action@v2
+        with:
+          working-directory: web-ui
+          config: pageLoadTimeout=100000,baseUrl=http://localhost:8080
+          build: yarn build
+          start: yarn start
```
Comment (mkimberlin) [inline .github/workflows/gradle-build-feature.yml:54]: if we are building and running a docker image above (and pointing to that running image), why do we need the yarn build and yarn start here?

File: server/build.gradle [hunk 2]
```diff
@@ -1,8 +1,9 @@
 plugins {
     id("com.github.johnrengelman.shadow") version "7.0.0"
-    id("io.micronaut.application") version "1.5.3"
+    id("io.micronaut.application") version "1.4.2"
```
Comment (mkimberlin) [inline server/build.gradle:3]: Was this downgrade intentional?

Comment (moshirim) [reply server/build.gradle]: It was unintentional - thanks for pointing that out. I reverted it

File: .github/workflows/gradle-build-feature.yml (hunk 1 above)
Comment (moshirim) [reply .github/workflows/gradle-build-feature.yml]: It works without it. I deleted it. Thanks!

**Comment by Bagurp** [comment]
Any chance we could squash some of the commits in this PR before merging it?

**Comment by mkimberlin** [comment]
The login test does not complete. It shows as "pending" in the log:
https://github.com/oci-labs/check-ins/runs/4084677350?check_suite_focus=true

**Comment by moshirim** [comment]
> The login test does not complete. It shows as "pending" in the log: https://github.com/oci-labs/check-ins/runs/4084677350?check_suite_focus=true

Yes that was intentional. I used `it.skip` so that it does not run. That test passes successfully in local but fails on remote because we are unable to access localhost 8080. I was planning on discussing that with you and how we can get around that problem when writing Cypress tests. I have created an issue for that as well. @mkimberlin

**Comment by moshirim** [comment]
> Any chance we could squash some of the commits in this PR before merging it?

Addressing that now.

**Comment by mkimberlin** [comment]
I think instead of localhost, you want to point it at the docker host. Try `host.docker.internal`

**Comment by mkimberlin** [comment]
If that doesn't work, then try `172.17.0.1`

**Pull request closed without merging**